packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icon.dart';
+import 'icon_theme.dart';
+import 'icon_theme_data.dart';
+import 'icons.dart';
+import 'list_tile.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+const Duration _kExpand = const Duration(milliseconds: 200);
+
+/// A single-line [ListTile] with a trailing button that expands or collapses
+/// the tile to reveal or hide the [children].
+///
+/// This widget is typically used with [ListView] to create an
+/// "expand / collapse" list entry.
+///
+/// See also:
+///
+///  * The "Expand/collapse" section of
+///    <https://material.io/guidelines/components/lists-controls.html>
+class ExpansionTile extends StatefulWidget {
+  /// Creates a single-line [ListTile] with a trailing button that expands or collapses
+  /// the tile to reveal or hide the [children].
+  const ExpansionTile({
+    Key key,
+    this.leading,
+    @required this.title,
+    this.backgroundColor,
+    this.onOpenChanged,
+    this.children: const <Widget>[],
+  }) : super(key: key);
+
+  /// A widget to display before the title.
+  ///
+  /// Typically a [CircleAvatar] widget.
+  final Widget leading;
+
+  /// The primary content of the list item.
+  ///
+  /// Typically a [Text] widget.
+  final Widget title;
+
+  /// Called when the tile expands or collapses.
+  ///
+  /// When the tile starts expanding, this function is called with the value
+  /// `true`. When the tile starts collapsing, this function is called with
+  /// the value false.
+  final ValueChanged<bool> onOpenChanged;
```

Comment:
ExpansionPanel has an "expansionCallback" which doesn't exactly jibe with flutter naming conventions. However given that this class is called ExpansionTile, renaming the callback onExpansionChanged makes sense.